File: src/libs/ThirdParty.VBSuppliers/Properties/ExpediaRapid/SerializableClasses/Book/BookResponse.vb
Imports Newtonsoft.Json.Linq
Imports Intuitive.Net
Imports Newtonsoft.Json

Namespace ExpediaRapid.SerializableClasses.Book

    Public Class BookResponse
        Implements IExpediaRapidResponse

        <JsonProperty("itinerary_id")>
        Public Property ItineraryID As String

        <JsonProperty("links")>
        Public Property Links As New Dictionary(Of String, Link)

        Public Function IsValid(responseString As String, statusCode As Integer) As Boolean Implements IExpediaRapidResponse.IsValid

            If Not String.IsNullOrWhiteSpace(responseString) Then
                Try
                    Dim bookResponse As BookResponse = JsonConvert.DeserializeObject(Of BookResponse)(responseString)

                    If String.IsNullOrWhiteSpace(bookResponse.ItineraryID) _
                        OrElse Not bookResponse.Links.ContainsKey("retrieve") Then Return False

                    Return True
                Catch ex As Exception
                    Return False
                End Try
            End If
            Return False
        End Function

    End Class

End Namespace

File: src/libs/ThirdParty.CSSuppliers/ExpediaRapid/SerializableClasses/Book/BookResponse.cs
namespace ThirdParty.CSSuppliers.ExpediaRapid.SerializableClasses.Book
{
    using System;
    using System.Collections.Generic;
    using Newtonsoft.Json;

    public class BookResponse : IExpediaRapidResponse
    {

        [JsonProperty("itinerary_id")]
        public string ItineraryID { get; set; }

        [JsonProperty("links")]
        public Dictionary<string, Link> Links { get; set; } = new Dictionary<string, Link>();

        public bool IsValid(string responseString, int statusCode)
        {

            if (!string.IsNullOrWhiteSpace(responseString))
            {
                try
                {
                    var bookResponse = JsonConvert.DeserializeObject<BookResponse>(responseString);

                    if (string.IsNullOrWhiteSpace(bookResponse.ItineraryID) || !bookResponse.Links.ContainsKey("retrieve"))
                        return false;

                    return true;
                }
                catch (Exception ex)
                {
                    return false;
                }
            }
            return false;
        }

    }

}